kernel/src/syscall/mod.rs

```diff
@@ -274,7 +274,16 @@ macro_rules! impl_syscall_nums_and_dispatch_fn {
                 $(
                     $num => {
                         $crate::log_syscall_entry!($name);
-                        $crate::syscall::dispatch_fn_inner!(args, ctx, user_ctx, $handler $args)
+                        use ostd::panic;
+                        use crate::Errno;
+                        use crate::error::Error;
+                        let result = panic::catch_unwind(|| {
+                            $crate::syscall::dispatch_fn_inner!(args, ctx, user_ctx, $handler $args)
```

Comment:
**Performance concern.** This is on the critical path of system call handling, adding an overhead to every system call. Could you run [the getppid system call benchmark](https://github.com/asterinas/asterinas/blob/main/test/benchmark/lmbench/process_getppid_lat/run.sh) in the LMbench to ensure that no noticeable overhead is introduced?

Reply:
The performance is listed in the table below:

| Status    | 1      | 2      | 3      | 4      | 5      | Avg     |
| --------- | ------ | ------ | ------ | ------ | ------ | ------- |
| Before PR | 1.0503 | 1.2945 | 1.0626 | 1.0630 | 1.0397 | 1.10202 |
| After PR  | 1.1891 | 1.1758 | 1.1610 | 1.1658 | 1.1574 | 1.16982 |

This PR will introduce about **10% overhead**...

```
$ make run
$ chmod +x /benchmark/lmbench/process_getppid_lat/run.sh
$ /benchmark/lmbench/process_getppid_lat/run.sh
# repeat last cmd for 5 times
```

Reply:
Thanks for the benchmark results! So I think we can conclude that we should try a different approach 😄